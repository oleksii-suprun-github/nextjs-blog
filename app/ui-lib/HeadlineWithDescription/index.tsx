import clsx from 'clsx';

function HeadlineWithDescription({
  headline,
  description,
  category,
}: {
  headline: string;
  description: string;
  category?: boolean;
}) {
  return (
    <div className="mb-12">
      <h1 className={clsx('mb-4 text-3xl font-bold text-gray-200', category && 'text-5xl')}>
        {headline}
      </h1>
      <p className={clsx(category && 'text-xl')}>{description}</p>
    </div>
  );
}

export default HeadlineWithDescription;
