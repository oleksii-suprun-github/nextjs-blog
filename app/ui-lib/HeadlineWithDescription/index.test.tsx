import { render, screen } from '@testing-library/react';
import { describe, it, expect } from 'vitest';
import HeadlineWithDescription from '.';

describe('HeadlineWithDescription Component', () => {
  it('should render correctly without link and match snapshot', () => {
    const { asFragment } = render(
      <HeadlineWithDescription headline="Test Headline" description="Test Description" />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render correctly with link and match snapshot', () => {
    const { asFragment } = render(
      <HeadlineWithDescription
        headline="Test Headline"
        description="Test Description"
        link="/test-link"
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render headline and description', () => {
    render(<HeadlineWithDescription headline="Test Headline" description="Test Description" />);

    const headlineElement = screen.getByText('Test Headline');
    const descriptionElement = screen.getByText('Test Description');

    expect(headlineElement).toBeInTheDocument();
    expect(descriptionElement).toBeInTheDocument();
  });

  it('should render the link when link prop is provided', () => {
    render(
      <HeadlineWithDescription
        headline="Test Headline"
        description="Test Description"
        link="/test-link"
      />,
    );

    const linkElement = screen.getByRole('link', { name: 'Go to category' });
    expect(linkElement).toBeInTheDocument();
    expect(linkElement).toHaveAttribute('href', '/test-link');
  });

  it('should apply category styles when category prop is true', () => {
    render(
      <HeadlineWithDescription
        headline="Test Headline"
        description="Test Description"
        category={true}
      />,
    );

    const headlineElement = screen.getByText('Test Headline');
    const descriptionElement = screen.getByText('Test Description');

    expect(headlineElement).toHaveClass('text-5xl');
    expect(descriptionElement).toHaveClass('text-xl');
  });
});
