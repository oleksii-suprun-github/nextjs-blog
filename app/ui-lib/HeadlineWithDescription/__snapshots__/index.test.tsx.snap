// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HeadlineWithDescription Component > should render correctly with link and match snapshot 1`] = `
<DocumentFragment>
  <div
    class="mb-12 flex flex-col items-start justify-between md:flex-row md:items-center"
  >
    <div>
      <h1
        class="mb-4 text-3xl font-bold text-gray-200"
      >
        Test Headline
      </h1>
      <p
        class="text-gray-300"
      >
        Test Description
      </p>
    </div>
    <a
      class="btn no-animation mt-6 bg-brand-purple text-gray-300 hover:bg-brand-purple md:mt-0"
      href="/test-link"
    >
      Go to category
    </a>
  </div>
</DocumentFragment>
`;

exports[`HeadlineWithDescription Component > should render correctly without link and match snapshot 1`] = `
<DocumentFragment>
  <div
    class="mb-12"
  >
    <h1
      class="mb-8 text-3xl font-bold text-gray-200"
    >
      Test Headline
    </h1>
    <p
      class=""
    >
      Test Description
    </p>
  </div>
</DocumentFragment>
`;
